controllers/defaults.go

```diff
@@ -0,0 +1,64 @@
+// Copyright Contributors to the Open Cluster Management project
+package controllers
+
+const (
+	default_API_CPURequest    = "25m"
+	default_API_MemoryLimit   = "1Gi"
+	default_API_MemoryRequest = "1Gi"
+
+	default_Indexer_CPURequest    = "25m"
+	default_Indexer_MemoryLimit   = "1Gi"
+	default_Indexer_MemoryRequest = "32Mi"
+
+	default_Collector_CPURequest    = "25m"
+	default_Collector_MemoryLimit   = "768Mi"
+	default_Collector_MemoryRequest = "64Mi"
+
+	default_Postgres_CPURequest    = "25m"
+	default_Postgres_MemoryLimit   = "4Gi"
+	default_Postgres_MemoryRequest = "128Mi"
+
+	default_API_Replicas       = 2
+	default_Indexer_Replicas   = 1
+	default_Collector_Replicas = 1
+	default_Postgres_Replicas  = 1
```

Comment:
- Why API defaults to 2?
- Can Postgres have more than 1 replica with our current config?

Reply:
We have 2 pods in search V1 - Going with that.
We cannot have more than 1 replica for search postgres in current config

Reply:
- In V1, the API deployment is the only one that honors the AvailabilityMode set in MCH, so that's why it defaults to 2.
- Indexer should do the same as API.
- Maybe we need to add the concept of `max_replicas` ?  Postgres and collector MUST only have 1 replica, so it will cause problems if user tries to override.  I believe this PR is ignoring replicas for those pods, that's ok, but need to make sure we addd a status message or it will be confusing.

Reply:
I will do default , to 1 for all 4 pods . We will have a validation for the ReplicaCount to be > than 0 . 
max_replicas - concept, needs some discussion.